src/main/resources/api/service_openapi.yaml

```diff
@@ -234,6 +234,72 @@ components:
           type: string
           nullable: true
           format: uuid
+    DataReferenceDescription:
+      type: object
+      properties:
+        referenceId:
+          description: The ID of the data reference
+          type: string
+          format: uuid
+        name:
+          description: The name of the data reference; used to refer to the reference
+          type: string
+        workspaceId:
+          description: The ID of the workspace containing the reference
+          type: string
+          format: uuid
+        resourceDescription:
+          description: Description of the workspace resource, if this is a controlled reference.
+          nullable: true
```

Comment:
actually after pulling this bit into my changes, it looks like `resourceDescription` needs to be defined as:

```
resourceDescription:
  description: Description of the workspace resource, if this is a controlled reference.
  nullable: true
  allOf:
  - $ref: '#/components/schemas/ResourceDescription'
```

apparently the `$ref` keyword obliterates sibling keywords, so the nullable property was getting ignored. this is the solution/workaround

per https://stackoverflow.com/questions/40920441/how-to-specify-a-property-can-be-null-or-a-reference-with-swagger

Reply:
Good catch! That's also good to know about nullable in general, I don't see other places in this API that it would cause issues (except maybe error code descriptions), but it'll probably come up again in the future